### PR TITLE
anim-controller: fix transitions sorting / minimize code duplication

### PR DESCRIPTION
**Pull request by kungfooman**
A sort function doesn't act on true/false, so this never sorted anything (copy/paste test for F12/DevTools):

```js
transitions = [
  {priority: 10},
  {priority: 50},
  {priority: 1},
  {priority: 2},
  {priority: 100},
  {priority: 3},
];
transitions.sort(function (a, b) {
    return a.priority < b.priority;
});
console.log(transitions.map(_ => _.priority));
```

Result: `[10, 50, 1, 2, 100, 3]` (its fixable by replacing `<` with `-`)

And then priority sorting is used in multiple places, so this PR reduces the code size/repetition of the same sorting function by just reusing functions (imo saving time and hopefully less bugs)

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/core/sort.js
Comment (willeastcott) [inline src/core/sort.js:5]: I think you need to remove this blank line or VS Code can't associate the code with the JSDoc block.

File: src/core/sort.js
Comment (willeastcott) [inline src/core/sort.js:5]: To prevent this from appearing in the API reference manual, you'll need to add `@ignore`.

File: src/core/sort.js
Comment (willeastcott) [inline src/core/sort.js:11]: And another `@ignore`.

File: src/core/sort.js
Comment (willeastcott) [inline src/core/sort.js:7]: Why does `cmpPriority` need to be exported. Should it be internal to this module?

**Comment by willeastcott** [comment]
LGTM. However, @mvaligursky and @ellthompson should approve before merging since this affects both animation and graphics subsystems.

**Comment by mvaligursky** [comment]
I looks great, but I would not mind some simple test of this in the test/core folder

**Review by ellthompson — APPROVED** [review]

File: src/core/sort.js
Comment (kungfooman) [reply src/core/sort.js:7]: I just like to keep things public for potential reuse, but technically it can be private (since its not being used elsewhere)

Comment (mvaligursky) [reply src/core/sort.js:7]: I'll merge this and hide cmpPriority in separate PR.

**Pull request merged**
